src/MoBi.Engine/Sbml/UnitDefinitionImporter.cs

```diff
@@ -146,45 +155,48 @@ public IDimension ConvertUnit(UnitDefinition unitDefinition)
                sbmlUnitDefinition.getScale(),
                _baseUnitsDictionary);
          }
+
          var unitName = combinedUnit.Name;
-         var dimensionAndUnit = _dimensionFactory.FindUnit(unitName);
+         var dimensionAndUnit = _moBiDimensionFactory.FindUnit(unitName);
          if (dimensionAndUnit.dimension != Constants.Dimension.NO_DIMENSION)
          {
             _sbmlInformation.MobiDimension[sbmlUnit] = dimensionAndUnit.dimension;
-            _unitConvertionDictionary.Add(sbmlUnit, new UnitConvertionInfo() { Dimension = dimensionAndUnit.dimension, Unit = dimensionAndUnit.unit, Rate = combinedUnit.Rate });
+            _unitConversionDictionary.Add(sbmlUnit, new UnitConversionInfo {Dimension = dimensionAndUnit.dimension, Unit = dimensionAndUnit.unit, Rate = combinedUnit.Rate});
             return dimensionAndUnit.dimension;
          }
 
-         _unitConvertionDictionary.Add(sbmlUnit, new UnitConvertionInfo() { Dimension = dimension, Unit = dimension.DefaultUnit, Rate = 1 });
+         _unitConversionDictionary.Add(sbmlUnit, new UnitConversionInfo {Dimension = dimension, Unit = dimension.BaseUnit, Rate = 1});
```

Comment:
I am pretty sure you mean BaseUnit and not DefaultUnit. We always need to convert to BaseUnit.

Reply:
Sure